server_src/utils/zip.utils.js

```diff
@@ -0,0 +1,79 @@
+const fs = require("fs");
+const path = require("path");
+const archiver = require("archiver");
+const Logger = require("../lib/logger.js");
+
+const logger = new Logger("OctoFarm-Server");
+
+const logPath = "./logs";
+
+async function checkIfZipFileExists(fileName) {
+  let fileExists = false;
+  if (fs.existsSync(fileName)) await deleteExistingZipFile(fileName);
+
+  return fileExists;
+}
+
+function deleteExistingZipFile(fileName) {
+  return fs.unlinkSync(fileName);
+}
+
+async function createZipFile(fileName, filePaths) {
+  // create a file to stream archive data to.
+  const currentZipFile = path.join(logPath, fileName);
+
+  // Make sure existing zip files have been cleared from the system before continuing.
+  await checkIfZipFileExists(currentZipFile);
+
+  const output = fs.createWriteStream(currentZipFile);
+  const archive = archiver("zip", {
+    zlib: { level: 9 }, // Sets the compression level.
+  });
+
+  // Listeners
+  archive.on("warning", function (e) {
+    logger.warning("Warning generated from zip function | ", e);
+  });
+
+  archive.on("error", function (e) {
+    throw e;
+  });
+
+  // Listen for file close and return path for download.
+  output.on("close", function () {
+    return currentZipFile;
+  });
+
+  // pipe archive data to the file
+  archive.pipe(output);
+
+  console.log(filePaths);
+
+  // // append a file from stream
```

Comment:
Dont forget to remove commented out code

Reply:
Aye they'll be gone don't worry